### DISCUSSION TITLE
q-popup-edit @cancel function call not working

### DISCUSSION THREAD

**Discussion by DFDevPriv** [quasarframework/quasar#14396] [General - Components / Directives / etc] [answered]
Hi all,

I'm not sure if I'm missing sth in my code, but I tried now for some hours to get it working without success.
I want to use the @cancel Event to call a function which sets a variable to a value, but the function gets not called. My template code looks like this: (the function is only outputting some text to the console for testing purposes)

`<q-td key="departmentName" :props="props">`
`{{ props.row.departmentName }}`
                    `<q-popup-edit`
                      `v-model="props.row.departmentName"`
                      `title="Abteilung anpassen"`
                      `buttons`
                      `label-set="Speichern"`
                      `:validate="validateQSelection"`
                      `@hide="validateQSelection(props.row.departmentName)"`
                      `v-slot="scope"`
                      `@save="callUpdateAsset(props.row.id)"`
                      `@cancel="testCancel()">`
                      `<q-select`
                        `v-model="scope.value"`
                        `filled`
                        `clearable`
                        `transition-show="flip-up"`
                        `transition-hide="flip-down"`
                        `:options="departmentChangeSelectOptions"`
                        `option-value="ccName"`
                        `option-label="ccName"`
                       `use-input`
                        `input-debounce="0"`
                        `@filter="filterDepartmentChangeSelect"`
                        `emit-value`
                        `map-options`
                        `:error="errorNewAsset"`
                        `:error-message="errorNewAssetDepartmentMessage"/>`
                   `</q-popup-edit>`
                  `</q-td>`
`

I also prepared a reproduction example on CodePen: https://codepen.io/kadev/pen/YzLNWwN

My expectation would be that the function gets executed, when I click the cancel button or press ESC.

Could anyone tell me / give me a hint what I'm doing wrong?

Kind regards,
Kadev

**Comment by pdanpdan** [comment, marked as answer]
In the codepen it gets called in all cases I tested. Please write a step-by-step list of what you do in the codepen

**Reply by DFDevPriv** [reply]
Oh my goodness... You're completely right. I had a wrong assumption, I expected the cancel event to be fired on each cancel regardless of if the value changed or not. Not sure why I thought the behaviour would be different to the @save one.
Sorry, for bothering...
